src/core/time.d

```diff
@@ -117,7 +117,7 @@ ulong mach_absolute_time();
 }
 
 //To verify that an lvalue isn't required.
-version (unittest) private T copy(T)(T t)
+version (DRuntimeUnittest) private T copy(T)(T t)
```

Comment:
I believe you need to keep this because [it is called from a templated unittest](https://github.com/dlang/druntime/blob/26fa565714d76df892eb327e98235aa854468766/src/core/time.d#L2199-L2213).

At least that's what the CI suggests:
```
/var/lib/buildkite-agent/builds/ci-agent-89c4044a-12c8-4cab-b953-c7a634bd4891-5/dlang/druntime/build/distribution/bin/../imports/core/time.d(2202,21): Error: undefined identifier `copy`, did you mean import `core`?
--
  | /var/lib/buildkite-agent/builds/ci-agent-89c4044a-12c8-4cab-b953-c7a634bd4891-5/dlang/druntime/build/distribution/bin/../imports/core/time.d(2210,16): Error: undefined identifier `copy`, did you mean import `core`?
  | /var/lib/buildkite-agent/builds/ci-agent-89c4044a-12c8-4cab-b953-c7a634bd4891-5/dlang/druntime/build/distribution/bin/../imports/core/time.d(2211,16): Error: undefined identifier `copy`, did you mean import `core`?
  | /var/lib/buildkite-agent/builds/ci-agent-89c4044a-12c8-4cab-b953-c7a634bd4891-5/dlang/druntime/build/distribution/bin/../imports/core/time.d(2212,25): Error: undefined identifier `copy`, did you mean import `core`?
  | /var/lib/buildkite-agent/builds/ci-agent-89c4044a-12c8-4cab-b953-c7a634bd4891-5/dlang/druntime/build/distribution/bin/../imports/core/time.d(2272,20): Error: undefined identifier `copy`, did you mean import `core`?
  | /var/lib/buildkite-agent/builds/ci-agent-89c4044a-12c8-4cab-b953-c7a634bd4891-5/dlang/druntime/build/distribution/bin/../imports/core/time.d(2283,13): Error: undefined identifier `assertApprox`
  | /var/lib/buildkite-agent/builds/ci-agent-89c4044a-12c8-4cab-b953-c7a634bd4891-5/dlang/druntime/build/distribution/bin/../imports/core/time.d(2004,18): Error: template instance `core.time.MonoTimeImpl!cast(ClockType)0` error instantiating


```

Reply:
I don't think that's actually a problem. I think that the `-version=DRuntimeUnittest` is somehow not passed to the compiler when unittests are compiled.

Reply:
Are the `Makefile`s used for testing exactly the ones that are located in `phobos/`? Are they different?

Reply:
For the record, I both of you are correct.
The root cause was that even though the function was used only from druntime's unit tests, they were nested inside the `MonoTime` template, which was instantiated from phobos, without passing `-version=CoreUnittest`. I solved this by guarding all nested (and undocumented) unit tests with `version (CoreUnittest)`.